"My Taxi Service" task

Sir-Beard opened the pull request:
This is first pull request in my repo Taxi Service.
Added spring connection. 
Further updates will be provided.

Sir-Beard committed:
Taxi service. Adding spring file

Sir-Beard committed:
Taxi service. Configuring Spring connection instead of self-written Injector.class.
Rewritten ConnectionUtil class(similar to AppConfig class) and ManufacturerDaoImpl class (with established connection in each method).

Rommelua commented on src/main/java/com/taxi/Main.java:
```diff
@@ -1,42 +1,44 @@
+        //CREATE
+        manufacturer.setName("Ajanta Group");
+        manufacturer.setCountry("India");
+        manufacturerDao.create(manufacturer);
```
```suggestion
        manufacturer = manufacturerDao.create(manufacturer);
```

Rommelua commented on src/main/java/com/taxi/Main.java:
```diff
@@ -1,42 +1,44 @@
+        System.out.println(manufacturer);
+
+        //UPDATE
+        manufacturer.setId(12L);
```
```suggestion
```

Rommelua commented on src/main/java/com/taxi/Main.java:
```diff
@@ -1,42 +1,44 @@
+        System.out.println(manufacturerDao.get(4L));
+
+        //DELETE
+        manufacturerDao.delete(12L);
```
```suggestion
        manufacturerDao.delete(manufacturer.getId());
```

Rommelua commented on src/main/java/com/taxi/lib/Injector.java:
```diff
@@ -1,126 +1,126 @@
-        return classes;
-    }
-}
+package com.taxi.lib;
```
delete **lib** package at all

Rommelua commented on src/main/java/com/taxi/util/ConnectionUtil.java:
```diff
@@ -0,0 +1,44 @@
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+
+@Configuration
+@PropertySource("classpath:application.properties")
+@ComponentScan(basePackages = {"com.taxi"})
```
```suggestion
```

Rommelua commented on src/main/java/com/taxi/util/ConnectionUtil.java:
```diff
@@ -0,0 +1,44 @@
+@PropertySource("classpath:application.properties")
+@ComponentScan(basePackages = {"com.taxi"})
+public class ConnectionUtil {
+    private static final String URL = "spring.datasource.url";
+    private static final String NAME = "spring.datasource.username";
+    private static final String PASSWORD = "spring.datasource.password";
```
```suggestion
    private static final String URL = "datasource.url";
    private static final String NAME = "datasource.username";
    private static final String PASSWORD = "datasource.password";
```

Rommelua commented on src/main/java/com/taxi/config/AppConfig.java:
```diff
@@ -0,0 +1,17 @@
+import org.springframework.core.env.Environment;
+
+@Configuration
+@PropertySource("classpath:db.properties")
```
```suggestion
@PropertySource("classpath:application.properties")
```

Rommelua commented on src/main/java/com/taxi/util/ConnectionUtil.java:
```diff
@@ -0,0 +1,44 @@
+    private static final String NAME = "spring.datasource.username";
+    private static final String PASSWORD = "spring.datasource.password";
+
+    private final Environment environment;
+
+    public ConnectionUtil(Environment environment) {
+        this.environment = environment;
+    }
```
we do not use it now

Rommelua commented on src/main/java/com/taxi/Main.java:
```diff
@@ -1,42 +1,44 @@
+
+    public static void main(String[] args) {
+        AnnotationConfigApplicationContext applicationContext
+                = new AnnotationConfigApplicationContext(ConnectionUtil.class);
```
```suggestion
                = new AnnotationConfigApplicationContext(AppConfig.class);
```

Rommelua replied:
not fixed

Sir-Beard committed:
Taxi service. Reworked some classes.
Delete lib package with Dao.java nad Injector.java classes.

Rommelua commented on src/main/java/com/taxi/util/ConnectionUtil.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.Properties;
+import org.springframework.context.annotation.ComponentScan;
+
+@ComponentScan(basePackages = {"com.taxi"})
```
```suggestion
@Component
```

Sir-Beard committed:
Taxi service. Task2. Commit #1. Added new classes, reworked Main.java class to invoke "services" instead of "dao" classes. 19:14 01.06.2023

Sir-Beard committed:
Taxi service. Task2. Commit #2. Reworked classes to involve spring, especially ConnectionUtil.java class. 0:51 07.06.2023

Rommelua commented on src/main/java/com/taxi/util/ConnectionUtil.java:
```diff
@@ -0,0 +1,38 @@
+            properties.load(new FileInputStream("src/main/resources/application.properties"));
+            connection = DriverManager.getConnection(properties.getProperty(url),
+                    properties.getProperty(name),
+                    properties.getProperty(password));
```
```suggestion
            connection = DriverManager.getConnection(url, name, password);
```
Идея состоит в том чтобы отказаться от считывания файла вручную)

Sir-Beard committed:
Taxi service. Task2. Commit #3.
1.Abandoning the idea of reading the file "application.properties" manually in ConnectionUtil.java class (removed "Properties" from the method).
2.Rewrote the annotations @Value so that they worked normally.
3.Cleaning ConnectionUtil.java class.
4.Cleaning AppConfig.java class (removed "Environment environment").
23:30 07.06.2023

Rommelua commented on src/main/java/com/taxi/util/ConnectionUtil.java:
```diff
@@ -0,0 +1,29 @@
+
+@Component
+public class ConnectionUtil {
+    @Value(value = "${datasource.url}")
```
```suggestion
    @Value("${datasource.url}")
```

Sir-Beard committed:
Taxi service. Task2. Commit #4.
Small edit in ConnectionUtil.java class.
Commit-push into add-spring branch.
21:57 08.06.2023
